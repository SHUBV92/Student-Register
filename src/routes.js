import Home from './containers/Home/Home';
import Register from './containers/Register/Register';
import Students from './containers/Students/Students';
import EditStudents from './containers/EditStudents/EditStudents';

export default [
  {
    path: '/',
    exact: true,
    name: 'Home',
    sidebar: () => Home,
    main: () => Home,
  },
  {
    path: '/students',
    name: 'Students',
    sidebar: () => Home,
    main: () => Students,
  },
  { path: '/students(/:id)', name: 'EditStudents', Component: EditStudents },
  {
    path: '/register',
    name: 'Register',
    sidebar: () => Home,
    main: () => Register,
  },
];
