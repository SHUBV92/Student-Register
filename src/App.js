import { useState } from 'react';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
import routes from './routes';
import StudentData from './studentsDataBase.json';

// styles
import { AppContainer, Nav, Logo } from './App.styles';

// components
import Home from './containers/Home/Home';
import Register from './containers/Register/Register';
import Students from './containers/Students/Students';
import EditStudents from './containers/EditStudents/EditStudents';

function App() {
  const [students, setStudents] = useState(StudentData);
  return (
    <Router>
      <AppContainer>
        <Nav>
          <Logo>Logo</Logo>
          <ul>
            <li>
              <Link to='/'>Home</Link>
            </li>
            <li>
              <Link to='/students'>Students</Link>
            </li>
          </ul>
        </Nav>

        {routes.map((route) => (
          <Route key={route.path} path={route.path} exact={route.exact}>
            <route.sidebar />
          </Route>
        ))}
        <div>
          {routes.map((route) => (
            <Route key={route.path} path={route.path} exact={route.exact}>
              <route.sidebar />
            </Route>
          ))}
        </div>
        {/* <Switch>
          {routes.map(({ path, Component }, key) => (
            <Route
              exact
              path={path}
              key={key}
              render={(props) => {
                const crumbs = routes
                  .filter(({ path }) => props.match.path.includes(path))
                  .map(({ path, ...rest }) => ({
                    path: Object.keys(props.match.params).length
                      ? Object.keys(props.match.params).reduce(
                          (path, param) =>
                            path.replace(
                              `:${param}`,
                              props.match.params[param]
                            ),
                          path
                        )
                      : path,
                    ...rest,
                  }));

                console.log(`Generated crumbs for ${props.match.path}`);
                crumbs.map(({ name, path }) => console.log({ name, path }));

                return (
                  <div className='p-8'>
                    <Component
                      {...props}
                      students={students}
                      setStudents={setStudents}
                    />
                  </div>
                );
              }}
            />
          ))}
        </Switch> */}
      </AppContainer>
    </Router>
  );
}

export default App;
