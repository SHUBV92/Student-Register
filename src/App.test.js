import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';
import routes from './routes';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

test('correct routes are exported', () => {
  expect(routes.map(({ name, path }) => `${name}: ${path}`)).toMatchSnapshot();
});

test('should render one student when user searches for shubinder', () => {
  render(<App />);

  let students = screen.getByRole('listitem');
  expect(students.length).toEqual(4);

  const searchBar = screen.getByRole('textbox');
  userEvent.type(searchBar, 'shubinder');

  students = screen.getAllByRole('listitem');
  expect(students.length).toEqual(1);
});
