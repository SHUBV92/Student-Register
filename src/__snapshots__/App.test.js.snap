// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct routes are exported 1`] = `
Array [
  "Home: /",
  "Students: /students",
  "EditStudents: /students(/:id)",
  "Register: /register",
]
`;
